src/main/java/com/flightstats/hub/app/ClusterHubBindings.java

```diff
@@ -55,58 +61,97 @@ protected void configure() {
         logger.info("starting server {} ", HubHost.getLocalName());
 
         bind(AwsConnectorFactory.class).asEagerSingleton();
-        bind(S3Config.class).asEagerSingleton();
+        bind(HubS3Client.class).asEagerSingleton();
+
         bind(LargeContentUtils.class).asEagerSingleton();
-        bind(WriteQueueConfig.class).toProvider(WriteQueueConfigProvider.class).asEagerSingleton();
-        bind(S3WriteQueue.class).asEagerSingleton();
-        bind(S3WriteQueueLifecycle.class).asEagerSingleton();
         bind(ContentService.class)
                 .to(ClusterContentService.class).asEagerSingleton();
-        bind(RemoteSpokeStore.class).asEagerSingleton();
-        bind(ContentDao.class)
-                .annotatedWith(Names.named(ContentDao.SINGLE_LONG_TERM))
-                .to(S3SingleContentDao.class).asEagerSingleton();
-        bind(ContentDao.class)
-                .annotatedWith(Names.named(ContentDao.BATCH_LONG_TERM))
-                .to(S3BatchContentDao.class).asEagerSingleton();
-        bind(ContentDao.class)
-                .annotatedWith(Names.named(ContentDao.LARGE_PAYLOAD))
-                .to(S3LargeContentDao.class).asEagerSingleton();
         bind(DynamoUtils.class).asEagerSingleton();
-        bind(S3BatchManager.class).asEagerSingleton();
-        bind(S3Verifier.class).asEagerSingleton();
         bind(AppUrlCheck.class).asEagerSingleton();
 
-        bind(SpokeTtlEnforcer.class)
-                .annotatedWith(Names.named(SpokeStore.WRITE.name()))
-                .toInstance(new SpokeTtlEnforcer(SpokeStore.WRITE));
-
-        bind(SpokeTtlEnforcer.class)
-                .annotatedWith(Names.named(SpokeStore.READ.name()))
-                .toInstance(new SpokeTtlEnforcer(SpokeStore.READ));
-
-        bind(DocumentationDao.class).to(S3DocumentationDao.class).asEagerSingleton();
         bind(SpokeDecommissionManager.class).asEagerSingleton();
-        bind(HubS3Client.class).asEagerSingleton();
-        bind(S3AccessMonitor.class).asEagerSingleton();
         bind(PeriodicMetricEmitter.class).asEagerSingleton();
         bind(PeriodicMetricEmitterLifecycle.class).asEagerSingleton();
+
+        if (HubProperties.isS3ConfigManagementEnabled()) {
+            bind(S3Config.class).asEagerSingleton();
+        }
+
+        if (HubProperties.isReadOnly()) {
+            bind(WriteQueue.class).to(NoOpWriteQueue.class).asEagerSingleton();
+        } else {
+            bind(WriteQueueConfig.class).toProvider(WriteQueueConfigProvider.class).asEagerSingleton();
+            bind(WriteQueue.class).to(S3WriteQueue.class).asEagerSingleton();
+            bind(S3WriteQueueLifecycle.class).asEagerSingleton();
+            bind(S3BatchManager.class).asEagerSingleton();
+            bind(S3Verifier.class).asEagerSingleton();
+            bind(S3AccessMonitor.class).asEagerSingleton();
+        }
+    }
+
+    @Inject
+    @Singleton
+    @Provides
+    @Named("WRITE")
```

Comment:
We may already have WRITE/READ public final strings defined somewhere that makes sense to use... Not entirely sure that it wouldn't overload the meaning if they do already exist, though,

Reply:
Yeah, /agree on all of this.  The unpleasantness is that SpokeStore is an enum (cool), but the Named annotation requires a constant val.  The fix is for an enum to expose its members's names separately as constant strings, which, eww.  Still, it's a bit safer.  Updating.
We use WRITE/READ quite a bit, yeah, but since this is specific to the SpokeStore, I didn't want to re-use constants from ContentService, etc.

Reply:
Yea, it's a bummer you can't use Enums with ```@Named```. I understand why, compile time vs runtime, blah, blah. It leads to exactly the hacks you mentioned, an Enum referencing a static String, but it's better than nothing.